user_guide.adoc

```diff
@@ -688,8 +685,8 @@ pattern over and over. Repetition might be:
 * element repeated while some condition is satisfied (or until some
 condition won't become true)
```

Comment:
To describe the `repeat-until` key, it should be:

```suggestion
* element repeated while some condition is not satisfied (or until some
condition becomes true)
```

Reply:
I am unclear on this...

Reply:
> I am unclear on this...

I don't know either how to formulate it, but the attribute is repeated as long as the `repeat-until` condition is `false`, i.e. while the condition is not satisfied. An example can be found below in the User Guide (https://doc.kaitai.io/user_guide.html#_repeat_until_condition_is_met):

```ksy
seq:
  - id: numbers
    type: s4
    repeat: until
    repeat-until: _ == -1
```

And it usually translates to this code:

```javascript
    do {
      var _ = this._io.readS4le();
      this.numbers.push(_);
    } while (!(_ == -1));
```

_`!` is a NOT operator._

And another way how to express "while the condition is not satisfied (`false`)" is "until the condition **is** satisfied (`true`)", because the word _until_ in English kind of means "while not ..." as I understand it.

So I think that _"until some condition won't become true"_ doesn't sound too English, because it can be rewritten to _"**while** some condition won't **not** become true"_, and we get a double negative here, which is not usual in English, and it apparently wasn't the intention anyway.

Reply:
My brain hurts, but I'll accept your suggestion anyway! It could always be improved another time if it needs it.